dev/com.ibm.ws.jdbc_fat_driver/publish/servers/com.ibm.ws.jdbc.fat.driver/server.xml

```diff
@@ -17,19 +17,31 @@
       <feature>jndi-1.0</feature>
     </featureManager>
  
-    <dataSource id="jdbc/derby" jndiName="jdbc/derby">
-      <jdbcDriver>
-        <library>
-          <fileset dir="${shared.resource.dir}/derby" includes="derby.jar"/>
-        </library>
-      </jdbcDriver>    
+ 
+    <library id="DerbyLib">
+        <file name="${shared.resource.dir}/derby/derby.jar"/>
+    </library>
+    
+    <library id="FATDriverLib">
+        <file name="${server.config.dir}/derby/FATDriver.jar"/>
+    </library>
+    
+    <dataSource id="derby" jndiName="jdbc/derby">
+      <jdbcDriver libraryRef="DerbyLib" />
       <properties databaseName="memory:jdbcdriver1" createDatabase="create" user="dbuser1" password="{xor}Oz0vKDtu" />
     </dataSource>
     
+    <dataSource id="fatDriver" jndiName="jdbc/fatDriver">
+ 	  <jdbcDriver libraryRef="FATDriverLib" />  
+      <properties url="jdbc:derby:memory:wrappedDerby;create=true" user="dbuser1" password="{xor}Oz0vKDtu" />
+    </dataSource>
+    
     <javaPermission codeBase="${shared.resource.dir}derby/derby.jar" className="java.security.AllPermission"/>
```

Comment:
Does the new driver also need permissions given that it includes Derby classes?

Reply:
Yes